### PR TITLE
npm package list support

### PR DESCRIPTION
**Pull request by tom--pollard**
This extends #99 to support npm, as part of the ongoing effort towards #83

### CODE REVIEW

**Comment by tom--pollard** [comment]
The behaviour of hard failing when hitting an `unpublished` package on the critical list seems sensible to me, however it should probably be aligned with how this is handled for the `pypi` implementation.

**Review by Qinusty — APPROVED** [review]
Minor point, otherwise looks good :+1:

**Comment by Qinusty** [comment]
> The behaviour of hard failing when hitting an `unpublished`(all existing versions removed, at the given point in time) package on the critical list seems sensible to me, however it should probably be aligned with how this is handled for the `pypi` implementation for a similar scenario. Any insight on this @Qinusty?

This could be quite problematic for a deployment with an NPM feed configured with many packages for critical polling. A single package being unpublished has the potential to prevent polling of all NPM packages specified.

**Comment by tom--pollard** [comment]
> > The behaviour of hard failing when hitting an `unpublished`(all existing versions removed, at the given point in time) package on the critical list seems sensible to me, however it should probably be aligned with how this is handled for the `pypi` implementation for a similar scenario. Any insight on this @Qinusty?
> 
> This could be quite problematic for a deployment with an NPM feed configured with many packages for critical polling. A single package being unpublished has the potential to prevent polling of all NPM packages specified.

Yes this is definitely a concern, it should be looked at as part of #107 and addressed along with other brittle areas

File: feeds/npm/README.md [hunk 1]
```diff
@@ -4,10 +4,15 @@ This feed allows polling of package updates from the repository.npmjs.org packag
 
 ## Configuration options
 
-The `packages` field is not supported by the npm feed.
-
+The `packages` Field can be supplied to the npm feed options to enable polling of package specific apis. This is less effective
```
[the remark this answers is not in the document]
Comment (tom--pollard) [reply feeds/npm/README.md]: Yeh it didn't read quite right to me either but couldn't put my finger on it at the time, cheers for the suggestion

**Pull request merged**